agent/agentssh/jetbrainstrack.go

```diff
@@ -38,30 +39,51 @@ func NewJetbrainsChannelWatcher(ctx ssh.Context, logger slog.Logger, newChannel
 		return newChannel
 	}
 
-	// If we do get a port, we should be able to get the matching PID and from
-	// there look up the invocation.
-	cmdline, err := getListeningPortProcessCmdline(d.DestPort)
+	// If we do get a port, we should be able to get the matching PID(s) and from
+	// there look up the invocation(s).  For now, ignore the address because we
+	// would need to resolve the address (for example `localhost` has a number of
+	// possibilities, or ::1 might route to ::, and so on).  The consequence is
+	// that if a user has another forwarded process listening on a different
+	// address but the same port as an active JetBrains process then the count
+	// will be inflated by however many processes are doing that.
```

Comment:
A change in architecture here would allow us to solve this problem.  Right now we're wrapping the normal handler, which means we don't have access to the actual network connection that gets dialed.  If instead we just had our own implementation, we could just dial the local socket and then see what address we connected to after the fact.  That means the process lookup is no longer ambiguous regarding v4 or v6.